src/navigation/navigation-events.js

```diff
@@ -97,9 +98,19 @@ function patchedUpdateState(updateState, methodName) {
     const urlAfter = window.location.href;
 
     if (!urlRerouteOnly || urlBefore !== urlAfter) {
-      window.dispatchEvent(
-        createPopStateEvent(window.history.state, methodName)
-      );
+      if (isStarted()) {
```

Comment:
I think the chances of it affecting someone are small, but still potentially there - this would be considered breaking-changes behavior right?

Reply:
Yeah there is a chance that this breaks something, similar to how 5.8.1 broke things in #655. The reason the artificial popstate events exist is for multiple single-spa applications to be able to have their own routers - my reasoning is that since no single-spa applications mount before start() is called, none of their routers will have set up their popstate listeners. And so any popstate listeners on the page before `start()` do not need those artificial popstates, since they exist separately from single-spa applications. However, there could be code unintentionally relying on those artificial popstates that are not single-spa applications.

I was thinking I'd release this PR as 5.8.2 instead of a major version, since the behavior being changed here is not documented. My resistance to publishing 6.0.0 is mostly just feelings that a 6.0.0 release would have to be accompanied by something substantially useful to users. Strictly following semver would say "mark it as 6.0.0," but "I don't want to explain to dozens of people that 6.0.0 is a really small change" is making me want to mark it as 5.8.2